lib/web_ui/dev/test_platform.dart

```diff
@@ -684,12 +684,21 @@ class BrowserManager {
   static Future<BrowserManager> start(
       Runtime runtime, Uri url, Future<WebSocketChannel> future,
       {bool debug = false}) {
+    final Stopwatch stopwatch = new Stopwatch();
+    final int timeoutMiliseconds = 30000;
+    stopwatch.start();
     var browser = _newBrowser(url, runtime, debug: debug);
 
     var completer = Completer<BrowserManager>();
 
     browser.onExit.then((_) {
-      throw Exception('${runtime.name} exited before connecting.');
+      // For the cases where we use a delegator such as `adb` (for Android) or
+      // `xcrun` (for IOS), these delegator processes can shut down before the
+      // websocket is available.
```

Comment:
I think we should remove `onExit` from `Browser`'s interface. If the `Browser` object does not actually represent the browser process, and its `onExit` may or may not indicate the browser exiting it becomes unreliable, and the warning messages misleading. Timeouts like this only lead to flakiness.

Instead, a concrete implementation of the `Browser` can internally perform these checks and handshakes as appropriate. Those implementations that launch the browser process directly will use the current method, and those using `adb` or `xcrun` will use something else.

Reply:
done (removed the on exit, just left comments there. I can also remove them if you think it's better)
Browser currently run checks on the exitCode and fails if it exists with non-zero code. I think that check should be enough. Since regardless of Chrome Browser of xcrun, nothing should
exit with a non-zero code.